cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceTaskContextProvider.java

```diff
@@ -172,11 +174,15 @@ public BasicMapReduceTaskContext load(ContextCacheKey key) throws Exception {
 
         TransactionSystemClient txClient = injector.getInstance(TransactionSystemClient.class);
 
+        Map<String, String> localDatasetNameMapping = contextConfig.getLocalDatasetNameMapping();
+        DatasetFramework dsFramework = localDatasetNameMapping == null ? datasetFramework
+          : new WorkflowDatasetFramework(datasetFramework, localDatasetNameMapping);
```

Comment:
This seems wrong as inside MapReduce it shouldn't be aware of workflow or local dataset at all. Seems like it should be done at the guice level.